tests/foreman/api/test_contentmanagement.py

```diff
@@ -125,6 +126,48 @@ def test_positive_sync_repos_with_lots_files(self):
         with self.assertNotRaises(TaskFailedError):
             repo.sync()
 
+    @tier2
+    def test_positive_kickstart_encoding_gzip(self):
+        """No encoding gzip errors on kickstart repositories
+        sync.
+
+        :id: dbdabc0e-583c-4186-981a-a02844f90412
+
+        :expectedresults: No encoding gzip errors present in /var/log/messages.
+
+        :CaseLevel: Integration
+
+        :steps:
+
+            1. Sync a kickstart repository.
+            2. After the repo is synced, change the download policy to
+                immediate.
+            3. Sync the repository again.
+            4. Assert that no errors related to encoding gzip are present in
+                /var/log/messages.
+
+        :CaseComponent: Pulp
+
+        :BZ: 1687801
+        """
+        org = entities.Organization().create()
+        product = entities.Product(organization=org).create()
+        repo = entities.Repository(
+            product=product,
+            url=CUSTOM_KICKSTART_REPO
+        ).create()
+        repo.sync()
+        repo = repo.read()
+        repo.download_policy = 'immediate'
+        repo = repo.update(['download_policy'])
+        repo.sync()
+        results = [
+            ssh.command('cat /var/log/messages | grep {}'.format(key))
+            for key in ('encoding', 'gzip', 'failed')
```

Comment:
Shouldn't we use some more specific terms? I can imagine some completely unrelated component might fail, polluting messages with `failed`, or some other process may add message about `encoding`. Maybe grep for `pulp.*{key}`?

Reply:
The error is related to the encoding of headers. I will work on improve those assertions.